### DISCUSSION TITLE
Maximum Speed/Bandwidth Supported by Strongswan

### DISCUSSION THREAD

**Discussion by jayaprakashsajja** [strongswan/strongswan#617] [Q&A]
I am having a Server with 40 Core and 256GB RAM with 40Gig SFP links.
Using Centos 7 with kernel 4.4 and installed Strongswan 5.7 but i am getting at a Max of 1.2 Gbps Speed/Bandwidth through Ipsec tunnel.
How can i optimize for more traffic ?
Any algorithm changes will make difference ?, if Yes what is the best algo we can use for ipsec to get Max traffic ?

**Comment by Thermi** [comment]
The only algorithm change can be by using something HW accelerated. For a single connection/flow, that's the only way. If you can use several connections in parallel, there are other ways to distribute them over several SAs and hence cores.

**Reply by jayaprakashsajja** [reply]
@Thermi  thanks for your response.
Several Connections corresponds to multiple Ipsec sessions ????
Do you believe that it may increase the whole IPsec Bandwidth  ?

**Reply by Thermi** [reply]
That's the trick on how to get more throughput.
A single SA can only be efficiently operated on by a single core because of the replay protection which requires serialization of the processing.

And it's not what I believe, it's something that people have tested, and are working on improving direct support in the kernel for.

**Reply by jayaprakashsajja** [reply]
@Thermi 
You saved my day, it worked, i can scale up my traffic/BW upto 60% of Link Speed now... Thanks...

**Reply by Thermi** [reply]
yw.

**Reply by Thermi** [reply]
BTW, you can probably cram out a lot more by optimizing the RSS, RPS, and RFS parameters used.

**Reply by jayaprakashsajja** [reply]
Any document for reference to derive optimum values for above said parameters ?????

**Reply by Thermi** [reply]
Found via google
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/performance_tuning_guide/sect-red_hat_enterprise_linux-performance_tuning_guide-networking-configuration_tools
